src/cmap/connection.ts

```diff
@@ -1103,76 +1102,87 @@ export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
       );
     }
 
-    let document = null;
+    let document;
     try {
-      document = await this.sendCommand(message, options);
-    } catch (ioError) {
-      if (this.monitorCommands) {
-        this.emit(
-          ModernConnection.COMMAND_FAILED,
-          new CommandFailedEvent(this as unknown as Connection, message, ioError, started)
-        );
-      }
-      throw ioError;
-    }
+      this.controller.signal.throwIfAborted();
+      for await (document of this.sendWire(message, options)) {
+        if (!Buffer.isBuffer(document) && document.writeConcernError) {
```

Comment:
That's correct, previously callback code "got away" with not checking for the type of document because of the lack of mutual exclusion between the error / document and just weak typing in general. For example, things like `document.writeConcernError` are just falsey if document is a Buffer. Now that we have better TS with promises we are forced to handle the potential for a Buffer, the code will continue to work as long as we pass it along to the next layer above. 

We should probably improve this, but I imagine it is a heavy lift that we will deal with in adopting on_demand bson anyway.